### PR TITLE
Create model for chat messages

### PR DESCRIPTION
**Pull request by ViktorHi**

### CODE REVIEW

File: src/main/java/com/example/demo/controllers/ChatController.java [hunk 1]
```diff
@@ -34,10 +32,8 @@ private synchronized Message getOne(@PathVariable String id){
 
     @PostMapping()
     private Message create (@RequestBody String mess){
-        long userId = Const.ADMIN_ID;
 
-        Message message = new Message(messageId.incrementAndGet(),
-                userId,
+        Message message = new Message(
                 mess,
                 new Date());
```
Comment (pashatyl) [inline src/main/java/com/example/demo/controllers/ChatController.java:38]: Better move timestamp into db or, at least, inside constructor

File: src/main/java/com/example/demo/controllers/ChatController.java [hunk 2]
```diff
@@ -57,7 +53,7 @@ private void delete (@PathVariable long id){
 
     private Message getMessage(long id){
         return messages.stream()
-                .filter(message -> message.getId()==id)
+                .filter(message -> message.getMessid()==id)
```
Comment (pashatyl) [inline src/main/java/com/example/demo/controllers/ChatController.java:56]: Do not use such shortening)

File: src/main/java/com/example/demo/dao/message/MessageDAOImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import com.example.demo.utils.*;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+import java.util.List;
+
+public class MessageDAOImpl implements MessageDAO {
+    @Override
+    public void save(Message message) {
+        Session session = HibernateSessionFactoryUtil.getSessionFactory().openSession();
```
Comment (pashatyl) [inline src/main/java/com/example/demo/dao/message/MessageDAOImpl.java:14]: Prefer using https://www.tutorialspoint.com/Automatic-resource-management-in-Java
Current implementation is not exception-secure

Comment (pashatyl) [reply src/main/java/com/example/demo/dao/message/MessageDAOImpl.java:14]: Session + transaction init may be moved to separated method (I'm wondering if it's not implemented in hibernate)

File: src/main/java/com/example/demo/services/MessageServices.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    void delete(Message message) {
+        messageDAO.delete(message);
+    }
+
+    List<Message> getAll(){
+        return messageDAO.getAll();
+    }
+
+
+    public static void main(String[] args) {
```
Comment (pashatyl) [inline src/main/java/com/example/demo/services/MessageServices.java:35]: Better use unit tests to cover functionality, not additional entry points

File: src/main/java/com/example/demo/services/MessageServices.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        MessageServices ms = new MessageServices();
+
+        Message hello = new Message("today", new Date());
+        System.out.println("message, before saving in db" + hello);
+
+        ms.save(hello);
+        System.out.println("saving success");
+
+        List<Message> all = ms.getAll();
```
Comment (pashatyl) [inline src/main/java/com/example/demo/services/MessageServices.java:44]:
```suggestion
        var all = ms.getAll();
```

File: src/main/java/com/example/demo/dao/message/MessageDAOImpl.java (hunk 3 above)
Comment (ViktorHi) [reply src/main/java/com/example/demo/dao/message/MessageDAOImpl.java:14]: do I understand correctly that the method should return an open session and transaction?

Comment (pashatyl) [reply src/main/java/com/example/demo/dao/message/MessageDAOImpl.java:14]: As for me, passing lambda action inside looks better

**Pull request merged**